packages/core/src/messages/responses/item-type-api.ts

```diff
@@ -0,0 +1,78 @@
+import {SdkError} from '../../errors/errors';
+import {
+  ResponseBase,
+  ResponseError,
+  ResponseHit,
+  ResponseMiss,
+} from './response-base';
+import {_ItemType} from './grpc-response-types';
+
+/**
+ * Parent response type for a item type request.  The
+ * response object is resolved to a type-safe object of one of
+ * the following subtypes:
+ *
+ * - {Hit}
+ * - {Miss}
+ * - {Error}
+ *
+ * `instanceof` type guards can be used to operate on the appropriate subtype.
+ * @example
+ * For example:
+ * ```
+ * if (response instanceof ItemType.Error) {
+ *   // Handle error as appropriate.  The compiler will smart-cast `response` to type
+ *   // `ItemType.Error` in this block, so you will have access to the properties
+ *   // of the Error class; e.g. `response.errorCode()`.
+ * }
+ * ```
+ */
+
+export abstract class Response extends ResponseBase {}
+
+class _Hit extends Response {
+  private readonly keyType: _ItemType;
```

Comment:
This is a value that an end user is going to have access to.  Therefore the type is something that the user will see and/or need to have access to in their code.  Therefore, it shouldn't start with an `_` and it shouldn't be defined in that `grpc-response-types` file, which contains internal classes that shouldn't be visible to users.

I think I would probably just define the user-facing enum directly in this file, or I would look and see where things like `CollectionTtl` are defined and put it in some file like that.